packages/jest-circus/src/utils.ts

```diff
@@ -332,20 +337,21 @@ export const makeSingleTestResult = (
   if (includeTestLocationInResult) {
     const stackLines = test.asyncError.stack.split('\n');
     const stackLine = stackLines[1];
-    let parsedLine = stackUtils.parseLine(stackLine);
-    if (parsedLine?.file?.startsWith(jestEachBuildDir)) {
+    let parsedLine = parseLine(stackLine);
+    if (
+      parsedLine &&
+      getAbsoluteSitePath(parsedLine)?.startsWith(jestEachBuildDir)
+    ) {
       const stackLine = stackLines[4];
-      parsedLine = stackUtils.parseLine(stackLine);
+      parsedLine = parseLine(stackLine);
     }
     if (
       parsedLine &&
-      typeof parsedLine.column === 'number' &&
-      typeof parsedLine.line === 'number'
+      parsedLine.type === 'CallSiteFrame' &&
+      parsedLine.callSite.site.type === 'FileSite'
```

Comment:
This is a good candidate for a sugar API, similar to the one I have provided for `getAbsoluteSitePath`. I still need to consider just how many such methods I want to provide and how.